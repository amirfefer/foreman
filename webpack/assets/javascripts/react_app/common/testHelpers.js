import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

export default {
  mockStorage: () => {
    const storage = {};

    return {
      setItem: (key, value) => {
        storage[key] = value || '';
      },
      getItem: key => (key in storage ? storage[key] : null),
      removeItem: (key) => {
        delete storage[key];
      },
      get length() {
        return Object.keys(storage).length;
      },
      key: (i) => {
        const keys = Object.keys(storage);

        return keys[i] || null;
      },
    };
  },
};

// a helper method for invoking a class method (for unit tests)
// obj = a class
// func = a tested function
// objThis = an object's this
// arg = function args

export const classFunctionUnitTest = (obj, func, objThis, args) =>
  obj.prototype[func].apply(objThis, args);

export const testComponentSnapshotsWithFixtures = (Component, fixtures) => {
  Object.entries(fixtures).forEach(([description, props]) =>
    it(description, () => {
      const wrapper = shallow(<Component {...props} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    }));
};
