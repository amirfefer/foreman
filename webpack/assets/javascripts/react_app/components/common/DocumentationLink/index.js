import React from 'react';
import { MenuItem } from 'patternfly-react';
import PropTypes from 'prop-types';
import Icon from '../Icon';

const DocumentationLink = ({ href, id }) => {
  const handleClick = e => {
    e.preventDefault();
    window.open(href, '_blank');
  };

  return (
    <MenuItem key="documentationUrl" id={id} href={href} onClick={handleClick}>
      <Icon type="question-sign" className="icon-black" />
      {` ${__('Documentation')}`}
    </MenuItem>
  );
};

export default DocumentationLink;

DocumentationLink.propTypes = {
  href: PropTypes.string.isRequired,
  id: PropTypes.string,
};

DocumentationLink.defaultProps = {
  id: 'documentationLink',
};
