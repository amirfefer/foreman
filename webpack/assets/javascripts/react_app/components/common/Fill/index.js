import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as actions from './FillActions';
import reducer from './FillReducer';

import Fill from './Fill';

// map state to props
const mapStateToProps = state => ({
  components: state.extendable,
});

// map action dispatchers to props
const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

// export reducers
export const reducers = { extendable: reducer };

// export connected component
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Fill);
