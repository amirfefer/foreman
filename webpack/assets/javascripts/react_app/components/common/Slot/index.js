import { connect } from 'react-redux';
import { selectRegisteredFills } from './SlotSelectors';
import Slot from './Slot';

// map state to props
const mapStateToProps = (state, ownProps) => ({
  fills: selectRegisteredFills(state, ownProps.id),
});

// export connected component
export default connect(mapStateToProps)(Slot);
