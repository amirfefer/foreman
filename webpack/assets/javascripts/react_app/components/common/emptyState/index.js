import React from 'react';
import { EmptyState as PfEmptyState, Button } from 'patternfly-react';

const EmptyState = (props) => {
  const {
    icon = 'add-circle-o',
    header,
    description,
    documentation: {
      label = __('For more information please see'),
      buttonLabel = __('Documentation'),
      url,
    },
    action,
    secondayActions,
  } = props;

  return (
    <PfEmptyState>
      <PfEmptyState.Icon type="pf" name={icon} />
      <PfEmptyState.Title>{header}</PfEmptyState.Title>
      <PfEmptyState.Info>{description}</PfEmptyState.Info>
      {url &&
        <PfEmptyState.Help>
         {label}
          <a href={url}> {buttonLabel} </a>
        </PfEmptyState.Help>
      }
      <PfEmptyState.Action>
        <Button href={action.url} bsStyle="primary" bsSize="large">
          {action.title}
        </Button>
      </PfEmptyState.Action>
      {secondayActions && (
        <PfEmptyState.Action secondary>
          {secondayActions.map(item => (
            <Button href={action.url} title={action.title}>
              {item.title}
            </Button>
          ))}
        </PfEmptyState.Action>
      )}
    </PfEmptyState>
  );
};

export default EmptyState;

