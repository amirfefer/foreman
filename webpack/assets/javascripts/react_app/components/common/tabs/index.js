import React from 'react';
import {
  TabContainer,
  Nav,
  NavItem,
  TabPane,
  TabContent,
} from 'patternfly-react';

const TabsWrapper = (props) => {
  const { tabs, children } = props;
  return (
    <TabContainer id="basic-tabs-pf" defaultActiveKey={0}>
      <div>
        <Nav bsClass="nav nav-tabs nav-tabs-pf" justified>
          {tabs.map((tab, index) => (
            <NavItem key={index} eventKey={index}>
              {tab}
            </NavItem>
          ))}
        </Nav>

        <TabContent animation>
          {React.Children.map(children, (content, index) => (
            <TabPane eventKey={index}>{content}</TabPane>
          ))}
        </TabContent>
      </div>
    </TabContainer>
  );
};

export default TabsWrapper;
