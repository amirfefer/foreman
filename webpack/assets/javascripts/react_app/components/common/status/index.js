import React from 'react';
import { connect } from 'react-redux';
import { get } from 'lodash';
import { Icon } from 'patternfly-react';
import { simpleLoader } from '../../common/Loader';
import * as StatusAction from '../../../redux/actions/status/';

export class Status extends React.Component {
  componentDidMount() {
    const { data: { id, type, url }, getStatus } = this.props;
    if (url) {
      getStatus({ id, type }, url);
    }
  }

  render() {
    const {
      status: {
        status, message, error, success,
      },
      getMessage,
    } = this.props;

    if (message && getMessage) {
      return <span>{message[getMessage]}</span>;
    }

    if (status || success !== undefined) {
      if (message && message.warning) {
        return (
          <div title={message.warning.message}>
            <Icon
              type='pf'
              name='warning-triangle-o'
            />
          </div>
        );
      }
      return (
        <div title={typeof message === 'string' ? message : undefined}>
          <Icon
            type='pf'
            name={status === 'OK' || success ? 'ok' : 'error-circle-o'}
          />
        </div>
      );
    }

    if (error) {
      return (
        <div title={message}>
          <Icon
            type='pf'
            name='error-circle-o' />
        </div>
      );
    }

    return simpleLoader('xs');
  }
}

const mapStateToProps = (state, ownProps) => {
  const { type, id } = ownProps.data;
  return {
    status: get(state.status, `${type}.${id}`) || {},
  };
};

export default connect(mapStateToProps, StatusAction)(Status);
