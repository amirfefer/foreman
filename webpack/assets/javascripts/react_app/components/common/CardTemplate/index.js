import React from 'react';
import CardItem from './CardItem';

import { addGlobalFill } from '../../common/Fill/GlobalFill';

const CardTemplate = ({ content, header, slotID, cardID }) => {
  addGlobalFill(
    slotID,
    cardID,
    <CardItem content={content} header={header} />,
    200
  );
};

export default CardTemplate;
