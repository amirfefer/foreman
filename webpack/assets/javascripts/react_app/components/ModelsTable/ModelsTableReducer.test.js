import {
  HW_MODEL_TABLE_REQUEST,
  HW_MODEL_TABLE_SUCCESS,
  HW_MODEL_TABLE_FAILURE,
} from './ModelsTableConsts';

import { testReducerSnapshotWithFixtures } from '../../common/testHelpers';
import reducer from './ModelsTableReducer';

const fixtures = {
  'should return initial state': {},
  'should handle HW_MODEL_TABLE_REQUEST': {
    action: {
      type: HW_MODEL_TABLE_REQUEST,
    },
  },
  'should handle HW_MODEL_TABLE_SUCCESS': {
    action: {
      type: HW_MODEL_TABLE_SUCCESS,
      payload: {
        search: 'name=model',
        results: [{ id: 23, name: 'model' }],
        page: 1,
        per_page: 5,
        total: 20,
        sort: { by: 'name', order: 'ASC' },
      },
    },
  },
  'should handle HW_MODEL_TABLE_FAILURE': {
    action: {
      type: HW_MODEL_TABLE_FAILURE,
      payload: {
        error: new Error('ooops!'),
      },
    },
  },
};

describe('ModelsTable reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
