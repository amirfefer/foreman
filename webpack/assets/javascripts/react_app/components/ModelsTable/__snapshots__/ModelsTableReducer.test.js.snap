// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelsTable reducer should handle HW_MODEL_TABLE_FAILURE 1`] = `
Object {
  "error": [Error: ooops!],
  "page": "1",
  "perPage": "20",
  "results": Array [],
  "searchQuery": null,
  "sortBy": "",
  "sortOrder": "",
  "status": "ERROR",
}
`;

exports[`ModelsTable reducer should handle HW_MODEL_TABLE_REQUEST 1`] = `
Object {
  "error": null,
  "page": "1",
  "perPage": "20",
  "results": Array [],
  "searchQuery": null,
  "sortBy": "",
  "sortOrder": "",
  "status": "PENDING",
}
`;

exports[`ModelsTable reducer should handle HW_MODEL_TABLE_SUCCESS 1`] = `
Object {
  "error": null,
  "page": 1,
  "perPage": 5,
  "results": Array [
    Object {
      "id": 23,
      "name": "model",
    },
  ],
  "searchQuery": "name=model",
  "sortBy": "name",
  "sortOrder": "ASC",
  "status": "RESOLVED",
}
`;

exports[`ModelsTable reducer should handle INIT_TABLE 1`] = `
Object {
  "error": null,
  "page": "3",
  "perPage": "5",
  "results": Array [],
  "searchQuery": "name=modelName1",
  "sortBy": "vendor_class",
  "sortOrder": "ASC",
  "status": "RESOLVED",
}
`;

exports[`ModelsTable reducer should handle TABLE_FAILURE 1`] = `
Object {
  "error": [Error: ooops!],
  "page": "1",
  "perPage": "20",
  "results": Array [],
  "searchQuery": null,
  "sortBy": "",
  "sortOrder": "",
  "status": "ERROR",
}
`;

exports[`ModelsTable reducer should handle TABLE_REQUEST 1`] = `
Object {
  "error": null,
  "page": "1",
  "perPage": "20",
  "results": Array [],
  "searchQuery": null,
  "sortBy": "",
  "sortOrder": "",
  "status": "PENDING",
}
`;

exports[`ModelsTable reducer should handle TABLE_SUCCESS 1`] = `
Object {
  "error": null,
  "page": 1,
  "perPage": 5,
  "results": Array [
    Object {
      "id": 23,
      "name": "model",
    },
  ],
  "searchQuery": "name=model",
  "sortBy": "name",
  "sortOrder": "ASC",
  "status": "RESOLVED",
}
`;

exports[`ModelsTable reducer should return initial state 1`] = `
Object {
  "error": null,
  "page": "1",
  "perPage": "20",
  "results": Array [],
  "searchQuery": null,
  "sortBy": "",
  "sortOrder": "",
  "status": "PENDING",
}
`;
